Adding better light controls for Tolino Vision 6 based on PR #520

hanakomisa opened the pull request:
I lifted the light control additions from https://github.com/koreader/android-luajit-launcher/pull/520, only slightly dusted it up to make it work with the current codebase. All credits to @dboleslawski .  

This comes with some caveats:
1. As the original PR stated, this needs the "Modify System Settings" permission toggled on for KOReader for the sliders to work. Though considering there are root only lighting controls as well as ADB based ones, I feel like this is an okay tradeoff. (Should modify the wiki to mention this?)
2. The Vision 6 has inverted warmth control, but there are no other known issues otherwise.
~~3. I do not have a Shine 4 personally to test, but I'm operating under the assumption that it will work just fine seeing how architecturally similar the Epos3/Vision6/Shine4 all are. Testers would be crucial here.~~


~~Additionally, I've gone ahead and added the missing info for the Tolino Shine 4, using the information from this comment https://github.com/koreader/koreader/issues/8482#issuecomment-2533638374 as well as this issue: https://github.com/koreader/android-luajit-launcher/issues/532 .~~

~~Seeing how the Epos 3 and Vision 6 already has proper e-ink support using NGL4 EPD (with some caveats that I'll get into), I feel like it's fair to add support to the Shine 4 as well. I'm not entirely sure if I've completely added everything though, so would appreciate extra eyes as I'm completely new to this and this is my first time ever really contributing to anything :lul:~~

Now, I don't know why `persist.mode.global` gets set to 8000 when the official Tolino app is launched, but as I mentioned here in https://github.com/koreader/android-luajit-launcher/issues/532#issuecomment-3239575034, using the AOSP settings app can reliably set it back to 0.

I also recently observed said prop being set to `1800010`, I have no clue how that happened and it's still something I'm looking into. It seems to have enabled the "A2" mode of the screen as contrast was quite poor and has a lot of ghosting, but responded very quickly.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/koreader/android-luajit-launcher/568)
<!-- Reviewable:end -->

Frenzie reviewed (approved):
Nothing jumps out to me, so if it's tested. ;-)

hanakomisa commented:
Forgot to add the Shine 4's ID under enum, whoops

hugleo commented:
Since is tested only for Vision 6 then add only Vision 6 in the new driver should be good to go.

hanakomisa commented:
> Since is tested only for Vision 6 then add only Vision 6 in the new driver should be good to go.

I have removed all of the Shine 4's support and removed the Epos 3 mapping. I'll see if I can get some testers for those devices but in the meantime this should satisfy all?

hugleo commented:
@pazos Since you've commented on the based PR #520 see if you're okay with it aside from the "Modify System Settings" permission being toggled on as OP mentioned.
